test/exceptions/src/assert_fail.d

```diff
@@ -158,6 +158,20 @@ void testTemporary()
     assert(Bad() == Bad());
 }
 
+void testEnum()
+{
+    struct UUID {
+        union
+        {
+            ubyte[] data;
+        }
+    }
+
+    ubyte[] data;
+    enum ctfe = UUID();
+    assert(ctfe.data == data);
```

Comment:
You want to test the output string. I noticed that your earlier addition doesn't check the output string either.  This can be done with the `test` function. See the other tests.

Reply:
I didn't test the output string because this is primarily a compilable test like https://github.com/dlang/druntime/blob/e7df7706c224ff6c3d684a52ce5528a500449c7a/test/exceptions/src/assert_fail.d#L131-L135

But i will add the output verfication

Reply:
Using `test` would not work as the enum member would be copied into the parameter of `test`

Reply:
We would need to add another function that unlike `test` accepts a whole block of code.

Reply:
Or just replicate the code inside the `test` function in this one case.

Reply:
Added